rfcs/react-components/convergence/custom-styling.md

```diff
@@ -0,0 +1,142 @@
+# RFC: Custom Styling
+
+Geoff Cox
+
+## Summary
+
+Fluent UI React v9 is built today to enforce adopting the Fluent 2 design styles.
+Partners creating component libraries cannot provide their customers the ability to deviate from Fluent 2 design
+like they could with v0, v7, or v8.
+We need to define a custom styling escape hatch and pattern for these partners.
+
+## Background
+
+Fluent UI React v9 strongly conforms to the Fluent 2 Design to provide consistency in component behavior, layout, and style. Most partners are expected to adopt the Fluent 2 Design and remove custom styling that deviates from the design direction for Office.
+
+v9 provides several mechanisms to customize components:
+
+- custom themes
+- custom design tokens
+- CSS classes & styles built with Griffel
+- component part replacement using slots
+- component recomposition using hooks
+
+Some of our partner teams are building component libraries on top of Fluent UI React.
+They have customers who need to diverge from the Fluent 2 design for both the partners components and our components.
+Our partners need a mechanism where their customers could customize the specific style elements of components
+(e.g. button border) without undue effort.
+
+Our current customization mechanisms have several limitations:
+
+- design tokens don't provide the granularity of customization needed
+- the CSS vars for each design token have a ceiling of ~300 before performance degrades
+- CSS classes create tight binding to style selector implementation or encounter CSS specificity issues.
+- slot replacement and CSS classes require code modification at each call site
+- recomposing every component requires signficant effort and creates a new component type that
+  must be adopted at each call site.
+
+These limitations are by design to either meet performance requirements, avoid bloating the API with kitchen-sink props,
+are limitations from target browsers or web languages/frameworks, or avoid past mistakes in previous versions.
+
+### Case: _Partner customer who needs centralized custom styling_
+
+//TODO: Mason and/or Brandon to fill in from admin-controls
+//TODO: Geoff to fill in power-platform custom
+
+## Problem statement
+
+We need to define a mechanism to allow:
+
+- consumers of Fluent UI React to centrally customize styles of components
```

Comment:
Consider re-phrasing for clarity:

- consumers of Fluent UI React to override the css of all the dom elements of a specified component within a given context.